File: isegoria/src/main/java/com/eulersbridge/isegoria/friends/ViewHolderDataSource.java
package com.eulersbridge.isegoria.friends;

import android.support.annotation.Nullable;
import android.support.v7.widget.RecyclerView;

import com.eulersbridge.isegoria.network.api.models.FriendRequest;

import java.lang.ref.WeakReference;

interface ViewHolderDataSource {
    void getFriendRequestInstitution(@Nullable Long institutionId, WeakReference<RecyclerView.ViewHolder> weakViewHolder);
    void onClick(FriendRequest friendRequest);
    void onActionClick(FriendRequest friendRequest);
}

File: isegoria/src/main/kotlin/com/eulersbridge/isegoria/friends/ViewHolderDataSource.kt
package com.eulersbridge.isegoria.friends

import android.support.v7.widget.RecyclerView

import com.eulersbridge.isegoria.network.api.model.FriendRequest

import java.lang.ref.WeakReference

internal interface ViewHolderDataSource {
    fun onClick(friendRequest: FriendRequest?)
    fun onActionClick(friendRequest: FriendRequest?)
}